crates/runtime/src/databackend/postgres.rs

```diff
@@ -142,6 +144,24 @@ impl PostgresBackend {
     }
 }
 
+fn read_pg_config(
+    params: Arc<Option<HashMap<String, String>>>,
+    secrets: Option<Secret>,
+) -> Arc<Option<HashMap<String, String>>> {
+    if let Some(params_val) = params.as_ref() {
+        let mut new_params = params_val.clone();
+        if let Some(secrets) = secrets {
+            if let Some(pg_pass_val) = new_params.get("pg_pass") {
+                if let Some(pg_pass_secret) = secrets.get(pg_pass_val) {
```

Comment:
If the secret doesn't exist, we should default to assuming we've been provided a raw password